src/Analyzer.TemplateProcessor/AnalyzerDeploymentsInterop.cs

```diff
@@ -13,6 +13,14 @@ private AnalyzerDeploymentsInterop()
 
         }
 
+        int IDeploymentsInterop.DeploymentNameLengthLimit => throw new System.NotImplementedException();
+
+        int IDeploymentsInterop.DeploymentKeyVaultReferenceLimit => throw new System.NotImplementedException();
+
+        int IDeploymentsInterop.DeploymentResourceGroupLimit => throw new System.NotImplementedException();
+
+        bool IDeploymentsInterop.KeyVaultDeploymentEnabled => throw new System.NotImplementedException();
```

Comment:
I can set them to these placeholder values: https://msazure.visualstudio.com/One/_git/AzureUX-Deployments?path=/src/DeploymentsCore/Instrumentation/DeploymentsInterop.cs&version=GBmaster&line=70&lineEnd=71&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents but now I'm having a hard time understanding why do we even define that AnalyzerDeploymentsInterop class, it doesn't seem to be used anywhere. I removed it, but can rever the commit if needed

Reply:
@yane3628 implemented it.  It seemed needed by the template parsing library.  Maybe the new version is different, but she has the most context on it.

Reply:
Also, to be clear, this class was used - it was initialized in the constructor of the Processor, which you removed.  DeploymentsInterop is a static class in the parser, and this class is simply what initialized that static class.  So the class is used, but the question is more - does the class it initializes need to be initialized?  That I'm not sure on.

Reply:
Yes, but I don't see any instances of AnalyzerDeploymentsInterop or DeploymentsInterop being used in the ArmTemplateProcessor, why would we need to initialize it?

Reply:
@JohnathonMohr do you think we should wait for Yanelis before merging? I'm tempted of merging anyways because no test was affected by this change

Reply:
I thought it might impact parsing certain types of templates not represented in our tests, but I just experimented with the Quick Start repo, and I get the same results with or without this Interop initialization.  I think it's safe to remove.